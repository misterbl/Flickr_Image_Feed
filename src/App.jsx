import React from "react";
import { ClipLoader } from "react-spinners";
import { connect } from "react-redux";
import * as apiThunk from "./actions/thunks/apiThunk";
import FlickrCard from "./components/FlickrCard";
import {
  getflickrFeedItems,
  fetchingFlickrFeed
} from "./selectors/apiSelectors";
import Header from "./components/Header";

export class App extends React.Component {
  setTimeoutId = 0;
  componentDidMount() {
    this.props.getflickrFeed();
  }

  render() {
    return (
      <>
        <Header />
        {this.props.fetchingFlickrFeed && (
          <div className="tc fixed mh7 pa7">
            <ClipLoader
              sizeUnit={"px"}
              size={40}
              color={"#123abc"}
              loading={true}
            />
          </div>
        )}
        <div className="flex flex-wrap ma3">
          {this.props.flickrFeedItems &&
            this.props.flickrFeedItems.map(item => (
              <FlickrCard
                image={item.media.m}
                title={item.title}
                author={item.author}
                author_id={item.author_id}
                link={item.link}
                description={item.description}
                tags={item.tags}
                key={item.title + item.published + item.media.m}
              />
            ))}
        </div>
      </>
    );
  }
}

export const mapStateToProps = state => ({
  flickrFeedItems: getflickrFeedItems(state),
  fetchingFlickrFeed: fetchingFlickrFeed(state)
});

export const mapDispatchToProps = {
  getflickrFeed: apiThunk.getflickrFeed,
  getflickrFeedByTags: apiThunk.getflickrFeedByTags
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
