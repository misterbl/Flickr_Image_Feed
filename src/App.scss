body {
  background-color: #f8f9fa;
  font-family: "Proxima Nova";
}
@media (max-width: 576px) {
  .form-container {
    margin: 0px;
  }
  body {
    background-color: white !important;
  }
}

.form-container {
  margin-left: 90px;
  margin-right: 90px;
  margin-top: 50px;
  margin-bottom: 50px;
}

.button-container {
  padding-right: 90px;
}

.back-button {
  border-color: #ced4da;
  width: 80px;
}

.next-button {
  background-color: #fc621b;
  color: white;
  width: 150px;
}

.copy-activity-button {
  background-color: #fc621b;
  color: white;
  min-width: 80px;
  font-size: 10px;
  height: 50px;
}

.close-modal-button {
  background-color: transparent;
  border: none;
  position: absolute;
  right: 10px;
  top: 10px;
  font-size: 19px;
}

.font-30 {
  font-size: 30px;
}

.font-20 {
  font-size: 20px;
}
