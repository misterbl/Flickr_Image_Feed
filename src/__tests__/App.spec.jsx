import * as React from "react";
import { shallow } from "enzyme";
import { App } from "../App";

describe("App", () => {
  const props = {
    flickrFeedItems: [
      {
        media: {
          m: "mediaTest",
          title: "titleTest",
          author: "authorTest",
          author_id: "123",
          link: "linkTest",
          description: "descriptionTest",
          tags: "tagsTest"
        }
      }
    ],
    fetchingFlickrFeed: false,
    getflickrFeed: jest.fn(),
    getflickrFeedByTags: jest.fn()
  };

  const wrapper = shallow(<App {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
