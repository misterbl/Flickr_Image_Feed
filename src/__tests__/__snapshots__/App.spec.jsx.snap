// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchingFlickrFeed={false}
    flickrFeedItems={
      Array [
        Object {
          "media": Object {
            "author": "authorTest",
            "author_id": "123",
            "description": "descriptionTest",
            "link": "linkTest",
            "m": "mediaTest",
            "tags": "tagsTest",
            "title": "titleTest",
          },
        },
      ]
    }
    getflickrFeed={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    getflickrFeedByTags={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        false,
        <div
          className="flex flex-wrap ma3"
        >
          <FlickrCard
            image="mediaTest"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FlickrCard
              image="mediaTest"
            />,
          ],
          "className": "flex flex-wrap ma3",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "NaNmediaTest",
            "nodeType": "function",
            "props": Object {
              "author": undefined,
              "author_id": undefined,
              "description": undefined,
              "image": "mediaTest",
              "link": undefined,
              "tags": undefined,
              "title": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          false,
          <div
            className="flex flex-wrap ma3"
          >
            <FlickrCard
              image="mediaTest"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FlickrCard
                image="mediaTest"
              />,
            ],
            "className": "flex flex-wrap ma3",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "NaNmediaTest",
              "nodeType": "function",
              "props": Object {
                "author": undefined,
                "author_id": undefined,
                "description": undefined,
                "image": "mediaTest",
                "link": undefined,
                "tags": undefined,
                "title": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
